The goal of this code is to find between AMZN, DPZ, BTC, and NFLX, which asset had the highest percentage growth from the first date to the last date? As someone who just started learning, I will comment after each line before & after writing the code to reflect on my thought process and outcome after the execution.



kagglehub is official python library that allows us to access/download dataset. In the next cell, I will import my dataset using kagglehub library. stored the path into an variable



percent : final price minus intial price * 100
            inital price

In [1]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("hershyandrew/amzn-dpz-btc-ntfx-adjusted-may-2013may2019")

print("Path to dataset files:", path)

c:\Users\admin\Desktop\python-for-data-science\venv\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Path to dataset files: C:\Users\admin\.cache\kagglehub\datasets\hershyandrew\amzn-dpz-btc-ntfx-adjusted-may-2013may2019\versions\1


it is worth noting that kagglehub does save the dataset locally on your 
machine (inside a cache folder like C:\Users\admin\.cache\kagglehub\), 
it does this automatically behind the scenes with a single line of code. 

This is different from a manual download because:
- You do not need to click, unzip, or move files yourself
- The path is returned directly into your variable, ready to use
- If you run the same code again, it loads from the local cache instantly 
  without re-downloading from the internet
- It works seamlessly on cloud environments like Google Colab or Kaggle 
  Notebooks where manual downloading is not possible

So while the files do end up stored locally, the entire process is handled 
automatically through code, making it faster, repeatable, and pipeline-friendly

This is because the cache folder allows us to store frequently used data 
locally on our system. Instead of taking 4.4 seconds to reach Kaggle's 
servers every time, kagglehub can now access the data almost instantly 
since it is already saved in the local cache folder.

Between AMZN, DPZ, BTC, and NFLX, which asset had the highest percentage growth from the first date to the last date?

In [32]:
import pandas as pd
import numpy as np

os is python module that helps python to communicate with os, allowing you to access/modify/manipulate files and folders. Therefore allowing different features to be useable such as os.listdir() in our case.

In [33]:
import os
os.listdir(path)

['portfolio_data.csv']

In [34]:
print(os.listdir(path)[0])

portfolio_data.csv


os.listdir(path) instructs operating system to look for what is inside path variable directory, then lists all exisiting files and folders inside that dir has and returns it.

dataframe(df) is a variable that stores data into tabulated form (rows & columns), os.path.join(folder, file) essentially this combines folder and file. Then, sep="," tells each column to seperate with "," .

In [35]:
df = pd.read_csv(os.path.join(path , "portfolio_data.csv"), sep=",")

In our case df variable stores the final date frame. folder and file path pd.read_csv reads the csv file then each row is seperated by /n and, seperates column.

In [36]:
type(df)

pandas.DataFrame

If we look at the type(df) of df, df is the object created, the class being DataFrame and type(df) displays the class  where df belongs to

pandas.DataFrame.

df.columns displays all the exisiting columns in the dataset.

In [37]:
df.columns

Index(['Date', 'AMZN', 'DPZ', 'BTC', 'NFLX'], dtype='str')

As you can see, each column name is separated by a comma. The data type of the column names is also being displayed as string.

In [38]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1520 entries, 0 to 1519
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    1520 non-null   str    
 1   AMZN    1520 non-null   float64
 2   DPZ     1520 non-null   float64
 3   BTC     1520 non-null   float64
 4   NFLX    1520 non-null   str    
dtypes: float64(3), str(2)
memory usage: 59.5 KB


From df.info() each data type of the rows are displayed, from eyeballing we can find the data type of date and NFLX is string but they should be int and float respectively.

In [39]:
df['Date'] = pd.to_datetime(df['Date'], format="mixed")

In [40]:
df['NFLX'] = pd.to_numeric(df['NFLX'].str.replace(' ', ''))

In [41]:
df.iloc[165]

Date    2013-12-24 00:00:00
AMZN             399.200012
DPZ               66.062965
BTC                  702.75
NFLX              54.055714
Name: 165, dtype: object

In [42]:
df['NFLX'] = pd.to_numeric (df['NFLX'])

In [43]:
df

,Date,AMZN,DPZ,BTC,NFLX
0,2013-05-01,248.229996,51.190983,106.250000,30.415714
1,2013-05-02,252.550003,51.987320,98.099998,30.641428
2,2013-05-03,258.049988,52.446388,112.900002,30.492857
3,2013-05-06,255.720001,53.205257,109.599998,30.098572
4,2013-05-07,257.730011,54.151505,113.199997,29.464285
...,...,...,...,...,...
1515,2019-05-08,1917.770020,283.149994,6171.959961,364.369995
1516,2019-05-09,1899.869995,282.160004,6358.290039,362.750000
1517,2019-05-10,1889.979980,278.369995,7191.359863,361.040009
1518,2019-05-13,1822.680054,273.880005,7980.129883,345.260010


In [44]:
df.tail(10)

,Date,AMZN,DPZ,BTC,NFLX
1510,2019-05-01,1911.520020,271.459991,5500.720215,378.809998
1511,2019-05-02,1900.819946,274.320007,5753.379883,379.059998
1512,2019-05-03,1962.459961,279.750000,5840.080078,385.029999
1513,2019-05-06,1950.550049,282.019989,5819.569824,378.670013
1514,2019-05-07,1921.000000,277.390015,5998.709961,370.459991
1515,2019-05-08,1917.770020,283.149994,6171.959961,364.369995
1516,2019-05-09,1899.869995,282.160004,6358.290039,362.750000
1517,2019-05-10,1889.979980,278.369995,7191.359863,361.040009
1518,2019-05-13,1822.680054,273.880005,7980.129883,345.260010
1519,2019-05-14,1840.119995,272.859985,8183.830078,345.609985


In [45]:
df.iloc[0]

Date    2013-05-01 00:00:00
AMZN             248.229996
DPZ               51.190983
BTC                  106.25
NFLX              30.415714
Name: 0, dtype: object

In [46]:
percentage_return_AMZN = (df.iloc[-1,1] - df.iloc[0,1]) / df.iloc[0,1] * 100

In [47]:
print(percentage_return_AMZN)

641.2963882898342


In [48]:
percentage_return_DPZ = (df.iloc[-1,2] - df.iloc[0,2]) / df.iloc[0,2] * 100

In [49]:
print(percentage_return_DPZ)

433.0235307261046


In [50]:
percentage_return_BTC = (df.iloc[-1,3] - df.iloc[0,3]) / df.iloc[0,3] * 100

In [51]:
print(percentage_return_BTC)

7602.428308705882


In [52]:
percentage_return_NFLX = (df.iloc[-1,4] - df.iloc[0,4]) / df.iloc[0,4] * 100

In [53]:
print(percentage_return_NFLX)

1036.2875946295392


In [54]:
if(percentage_return_AMZN >= percentage_return_BTC >= percentage_return_DPZ >= percentage_return_NFLX):
    print(percentage_return_AMZN, "gave the highest percentage return from 2013-05-01 to 2013-05-01")

elif(percentage_return_BTC >= percentage_return_DPZ >= percentage_return_NFLX):
    print(percentage_return_BTC, "gave the highest percentage return from 2013-05-01 to 2013-05-01")

elif(percentage_return_DPZ >= percentage_return_NFLX):
    print(percentage_return_DPZ, "gave the highest percentage return from 2013-05-01 to 2013-05-01")

else:
    print(percentage_return_NFLX, "gave the highest percentage return from 2013-05-01 to 2013-05-01")

1036.2875946295392 gave the highest percentage return from 2013-05-01 to 2013-05-01


Between May 2013 and May 2019, which asset — AMZN, DPZ, BTC, or NFLX — had the most violent day to day price swings, which one was the calmest, and on what specific trading day did each asset experience its single biggest move compared to the previous session?

For this i would need to calculate percentage return for each trading session compared to the other day?

I will need to create a for loop and check for each row returns and in a loop 1520 rows and i need to run it so for eg: for(i=0;i<=1519.i++)

In [55]:
daily_returns=df[['AMZN','BTC','DPZ','NFLX']].pct_change()

In [56]:
return_per_ses_AMZN=df['AMZN'].pct_change()

In [57]:
return_per_ses_BTC=df['BTC'].pct_change

In [58]:
return_per_ses_DPZ=df['DPZ'].pct_change

In [59]:
return_per_ses_NFLX=df['NFLX'].pct_change

In [61]:
daily_returns.describe()

,AMZN,BTC,DPZ,NFLX
count,1519.000000,1519.000000,1519.000000,1519.000000
mean,0.001499,0.005838,0.001228,0.001955
std,0.018981,0.104611,0.015845,0.026728
min,-0.109972,-0.572057,-0.100809,-0.193696
25%,-0.007043,-0.014066,-0.006490,-0.010871
50%,0.001168,0.001969,0.001444,0.000660
75%,0.010868,0.021830,0.009273,0.014554
max,0.141311,3.368390,0.130006,0.190281


In [ ]:
daily_returns.iloc[2,1 ]

np.float64(0.15086650664355772)